### ISSUE TITLE
Zero-byte dbm file on save

### ISSUE THREAD

**Issue by pbryant-nectar** [pgmodeler/pgmodeler#1295]
**Bug description**
I have a dbm file that has two schemas defined.  I want to delete everything associated with one of those schemas.  When I try to Del. Cascade the entire schema the pgModeler app crashes.  That happens on 0.9.1 and 0.9.2-beta.  So I try to Del. Cascade each of the schema objects.  After doing that (not sure how far through the problem occurs) and then saving, you get a zero-byte file.

**How to reproduce**
Del. Cascade on many objects associated with one schema within my model.

**Expected behavior**
Valid dbm file.

**Screenshots**

**Info about your desktop**
 - OS: Macs
 - Version: 10.12.6 and 10.14.2
 - Window manager:
 - pgModeler version: 0.9.1 and 0.9.2-beta
 - Qt version:

**Stacktrace / Debug info**
(Although I refer to a reliable application crash in the bug description I will raise that as a separate issue)

**Comment by rkhaotix** [comment]
Hi @pbryant-nectar 

I've tested here on my development environment (openSuse Linux + KDE) and everything runs ok! Tested 3 different models and not a single crash happened. Maybe this is something particular to your model. Is there a possibility for you to share it so I can debug the application? If there's no such possibility, can you create a sample model in which the problem appears?

**Comment by pbryant-nectar** [comment]
Hi @rkhaotix 

I've shared a screencast and copy of the offending dbm file to you directly.

For the benefit of others, the problem has become worse than described above.  The worst characteristics of the problem include:

- Any time you make any change to the diagram (eg just moving an object on the diagram) the file seems to get auto-saved (even tho auto-save is disabled).
- When that happens, the auto-saved file is almost always zero bytes.
- You can work-around that by save-as a new filename (which always saves correctly as non-zero bytes, even though nothing else has changed)
- However as soon as you make any change to the diagram (eg just moving an object) the resulting auto-save causes a zero-byte file again.
- When I try to open the file in v0.9.1 it complains that the model needs fixing, but it fails to fix the model.
 
So it is impossible to work on the file because it is constantly saving, and what it saves is a zero-length file.

**Comment by rkhaotix** [comment]
Hi Paul. Thanks for such details.

According to what you've described this issue seems to be a regression and happens only on macOs. I'll take care of it ASAP during the next week.

**Comment by rkhaotix** [comment]
@pbryant-nectar I've fixed the problem with unstoppable savings on the model. The problem was happening when you deactivated the auto saving feature. Once doing that the timer that controls that saving was working on the opposite way... instead of stop (or never timing out) it was timing out when every event that the application was dispatching, this is why the software was unusable.

Now for the zero-byte bug, I couldn't reproduce it even with all the models you've sent me. I deleted a bunch of objects in cascade mode until the model had no object and no crash occurred. I'll keep investigating.
